ush/python/pygfs/task/analysis.py

```diff
@@ -200,6 +201,74 @@ def link_jediexe(self) -> None:
 
         return
 
+    @logit(logger)
+    def get_ens_dict(self, task_config: Dict[str, Any]) -> Dict[str, List[str]]:
+        """Compile a dictionary of ensemble member restarts to copy
+
+        This method constructs a dictionary of ensemble FV3 restart files (coupler, core, tracer)
+        that are needed for global atmens DA and returns said dictionary for use by the FileHandler class.
+
+        Parameters
+        ----------
+        config: Dict
+            a dictionary containing all of the configuration needed
+
+        Returns
+        ----------
+        ens_dict: Dict
+            a dictionary containing the list of ensemble member restart files to copy for FileHandler
+        """
+        # NOTE for now this is FV3 restart files and just assumed to be fh006
+
+        # define template
+        template_res = self.task_config.COM_ATMOS_RESTART_TMPL
+        prev_cycle = self.task_config.previous_cycle
+        tmpl_res_dict = {
+            'ROTDIR': self.task_config.ROTDIR,
+            'RUN': self.task_config.RUN,
+            'YMD': to_YMD(prev_cycle),
+            'HH': prev_cycle.strftime('%H'),
+            'MEMDIR': None
+        }
+
+        # set directory type based on RUN
+        if self.task_config.RUN in ['enkfgdas', 'enkfgfs']:
+            dirtype = 'bkg'
+        else:
+            tmpl_res_dict['RUN'] = 'enkf' + self.task_config.RUN
+            dirtype = 'ens'
```

Comment:
Now that I have had a chance to look at this, I would suggest turning this method in to a `staticmethod` and pass everything that this method needs through `config`, rather than relying on the global variable `self.task_config`.
This would also eliminate this `if-else` and raise it to the class that defines this specific need.

Reply:
I think I see what you are saying.  Let me see if I can correctly implement this suggestion.

Reply:
`get_fv3ens_dict` has been changed to a `staticmethod`.   The `if-else` in `get_fv3ens_dict` has been removed and necessary additions made in the classes which invoke the method.   Done at [`51200a0`](https://github.com/RussTreadon-NOAA/global-workflow/commit/51200a0d9459f9c67c45bd6cb284163603ca51e7).